src/abstractdataframe/reshape.jl

```diff
@@ -48,54 +51,68 @@ d1s_name = stack(d1, Not([:a, :b, :e]), variable_name=:somemeasure)
 """
 function stack(df::AbstractDataFrame, measure_vars::AbstractVector{<:Integer},
                id_vars::AbstractVector{<:Integer}; variable_name::Symbol=:variable,
-               value_name::Symbol=:value, view::Bool=false)
+               value_name::Symbol=:value, view::Bool=false, stringvar::Bool=false)
     if view
-        return _stackview(df, measure_vars, id_vars,
-                          variable_name=variable_name, value_name=value_name)
+        return _stackview(df, measure_vars, id_vars, variable_name=variable_name,
+                          value_name=value_name, stringvar=stringvar)
     end
     N = length(measure_vars)
-    cnames = names(df)[id_vars]
-    insert!(cnames, 1, value_name)
-    insert!(cnames, 1, variable_name)
-    DataFrame(AbstractVector[repeat(_names(df)[measure_vars], inner=nrow(df)), # variable
+    cnames = _names(df)[id_vars]
+    pushfirst!(cnames, value_name)
+    pushfirst!(cnames, variable_name)
+    if stringvar
+        nms = String.(_names(df)[measure_vars])
+        catnms = categorical(nms)
+        levels!(catnms, nms)
+    else
+        Base.depwarn("`stringvar=false` as a default is deprecated." *
```

Comment:
Usually we use `nothing` as a default so that people can get the old behavior without a deprecation. Or if we want to remove that possibility in the future, better say it.

BTW, there are three possibly useful types: categorical, symbol and string. Maybe the argument could take the type instead of being a Boolean?

Reply:
I will reply here to both comments as they are related. I think it is OK to allow either `Symbol`, `String` or categorical, with the default to categorical.

Maybe we should call kwarg `vartype` and allow it to take the following values:
* `:symbol`
* `:string`
* `:categorical` (this will be the default after the deprecation)

Now the ability to handle the deprecation of kwargs is not clean in Julia unfortunately. We could go for `nothing` as a temporary default as you propose (and not document it) and not document it (so that we can later remove it without deprecation).

Is what I propose OK?

Reply:
Why not take types rather than symbols? That sounds more Julian.

Regarding the deprecation, if we want to keep the current behavior in the next version and only change it in the following one, better use `nothing`. But we could also decide that it's not too breaking and just change the default now so that we don't have to wait for another release.